<?xml version="1.0"?>
<launch>
  <arg name="node_name" default="bb1_robot" />

  <!-- Use udev rules to setup aliases for right/left wheel port. -->
  <arg name="front_right_wheel_port" default="/dev/esc_front_right" />
  <arg name="back_right_wheel_port" default="/dev/esc_back_right" />
  <arg name="front_left_wheel_port" default="/dev/esc_front_left" />
  <arg name="back_left_wheel_port" default="/dev/esc_back_left" />
  
  <!-- Parameter for enabling rotor position publishing
       Possible values are "inductance", "observer", "encoder", "pid_pos", "pid_pos_error", "encoder_observer_error" -->
  <arg name="rotor_position_source" default="none" />

  <!-- Optionally launch manager in GDB, for debugging -->
  <arg name="debug" default="false" />
  <arg if="$(arg debug)" name="launch_prefix" value="xterm -e gdb --args" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <!-- Also globally disable bond heartbeat timeout in debug mode, so everything
       doesn't die when you hit a break point -->
  <param if="$(arg debug)" name="/bond_disable_heartbeat_timeout" value="true" />

    <node pkg="bb1_hw" type="robot" name="$(arg node_name)" output="screen" launch-prefix="$(arg launch_prefix)" >
      <param name="front_right_wheel_port" value="$(arg front_right_wheel_port)" />
      <param name="back_right_wheel_port" value="$(arg back_right_wheel_port)" />
      <param name="front_left_wheel_port" value="$(arg front_left_wheel_port)" />
      <param name="back_left_wheel_port" value="$(arg back_left_wheel_port)" />
      <param name="front_right_wheel_ikv" value="8.1" />
      <param name="back_right_wheel_ikv" value="8.1" />
      <param name="front_left_wheel_ikv" value="8.1" />
      <param name="back_left_wheel_ikv" value="8.1" />
      <param name="tacho_pulses_per_revolution" value="90" />
      <param name="motor_poles" value="15" />
      <param name="rotor_position_source" value="$(arg rotor_position_source)" />
    </node>

  <!-- Get the PID and velocity filtering for the motors -->
  <rosparam command="load" file="$(find bb1_hw)/config/myparams.yaml" />
</launch>
